ContactDetailsApi.Tests/V2/E2ETests/Stories/FetchAllContactDetailsTests.cs

```diff
@@ -56,4 +64,24 @@ public void ServiceReturnsAllContactDetailsAsRequested()
                 .BDDfy();
         }
     }
+
+    [Collection("AppTest collection")]
+    public class FetchAllContactDetailsTestsUnauthorised : FetchAllContactDetailsTests
+    {
+        public FetchAllContactDetailsTestsUnauthorised(
+            MockWebApplicationFactory<Startup> appFactory)
+        : base(appFactory)
+        {
+        }
+
+        [Fact]
+        public void ServiceReturns401Unauthorised()
+        {
+            this.Given(g => _contactDetailsFixture.GivenAFetchAllContactDetailsByUprnRequest())
+                .When(w => _steps.WhenAllContactDetailsAreRequested())
+                .Then(t => _steps.ThenAnUnauthorisedResponseIsReturned())
+                .BDDfy();
+        }
+    }
+
```

Comment:
Should this specify why an unauthorized response was returned? IE - because the ip address wasnt whitelisted